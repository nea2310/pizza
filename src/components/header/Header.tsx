import { useState } from 'react';
import { Link, NavLink } from 'react-router-dom';
import './header.scss';
import UserInfo from '../user-info/user-info';
import { useAppSelector } from '../../hooks';
import { IStore } from '../../interface';
import { routesMap } from '../../routes/routes';

interface IRoutesMap {
  [index: string]: { name: string; };
}

export default function () {
  const props: IStore = useAppSelector(state => {
    return state;
  });

  const [details, setDetails] = useState(false);

  const contentClassName = details ? 'header__content_active' : '';
  const contentClasses = ['header__content', contentClassName];

  const burgerClassName = details ? 'header__burger-button_active' : '';
  const burgerClasses = ['header__burger-button', burgerClassName];

  const name = props.user.user.name;
  const surname = props.user.user.surname;
  const isAuthed = !!props.user.user.token;
  const cartImg = require('./img/cart-shopping.svg').default;



  /*в зависимости от того, авторизован ли пользователь, отображаем либо приветстсвие, 
либо форму для ввода логина и пароля*/
  const loginForm =
    isAuthed ? (
      <div className='header__auth'>
        <UserInfo
          username={name}
          usersurname={surname}
        ></UserInfo>
      </div >
    ) :
      (<div className='header__auth'>
        <span><Link className='header__auth-link' to={'/login'}>Войти</Link></span>
        <span className='header__auth-text'>Еще не зарегистрированы?</span>
        <span>
          <Link className='header__auth-link' to={'/register'}>Зарегистрироваться</Link></span>
      </div>);

  const cart = (
    <NavLink
      to={routesMap.cart}>
      <div className='header__cart'>
        <img
          className='header__cart-img'
          src={cartImg} alt="корзина" />
      </div>
    </NavLink>
  );


  /*основное меню*/

  // eslint-disable-next-line no-undef
  const menuItems: JSX.Element[] = [];
  const pizzaItemsIMG = require('./img/logo.png');
  const list: IRoutesMap =
  {
    about: { name: 'О нас' },
    deals: { name: 'Акции' },
    blog: { name: 'Блог' },
  };
  for (let key in routesMap) {
    if (key in list) {
      const name = list[key].name;
      const menuItem = (
        <li
          className='header__nav-item'
          key={routesMap[key]}>
          <NavLink
            className='header__nav-link'
            to={routesMap[key]}>
            {name}
          </NavLink>
        </li>
      );
      menuItems.push(menuItem);
    }
  }

  return (
    <header className='header'>
      <button
        aria-label="навигация"
        className={burgerClasses.join(' ')}
        onClick={() => setDetails(prev => !prev)}
      >
        <div
          className='header__burger'>
        </div>
      </button>
      <div className='header__wrapper'>
        <div className='header__logo'>
          <NavLink
            className='header__nav-link'
            to={routesMap.pizzaItems}>
            <img
              className='header__nav-img'
              src={pizzaItemsIMG} alt="" />
          </NavLink>
        </div>
        <div className={contentClasses.join(' ')}>
          <nav className='header__nav'>
            <ul className='header__nav-list'>
              {menuItems}
            </ul>
          </nav>
          <div className='header__user'>
            {loginForm}
            {cart}
          </div>
        </div>
      </div>
    </header>
  );
};



