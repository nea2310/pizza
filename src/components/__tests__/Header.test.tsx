import Header from '../header/Header';
import { renderWithProviders } from '../../shared/testUtils/testUtils';
import { initialStore } from '../../shared/testUtils/mockedStore';
import { BrowserRouter } from 'react-router-dom';

describe('Header component rendering', () => {
  it('Renders Header component', async() => {
    const headerComponent = renderWithProviders(<BrowserRouter><Header /></BrowserRouter>, {
      preloadedState: initialStore
    });

    expect(headerComponent).toMatchSnapshot();
  });
});
