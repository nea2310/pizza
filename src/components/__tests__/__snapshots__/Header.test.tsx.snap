// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component rendering Renders Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <button
          aria-label="главное меню"
          class="header__burger-button "
        >
          <div
            class="header__burger"
          />
        </button>
        <div
          class="header__wrapper"
        >
          <div
            class="header__logo"
          >
            <a
              class="header__nav-link"
              href="/"
              title="главная страница"
            >
              <img
                alt="главная страница"
                class="header__nav-img"
                src="logo.png"
              />
            </a>
          </div>
          <div
            class="header__content "
          >
            <nav
              class="header__nav"
            >
              <ul
                class="header__nav-list"
              >
                <li
                  class="header__nav-item"
                >
                  <a
                    class="header__nav-link"
                    href="/about"
                  >
                    О нас
                  </a>
                </li>
                <li
                  class="header__nav-item"
                >
                  <a
                    class="header__nav-link"
                    href="/deals"
                  >
                    Акции
                  </a>
                </li>
                <li
                  class="header__nav-item"
                >
                  <a
                    class="header__nav-link"
                    href="/blog"
                  >
                    Блог
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="header__user"
            >
              <div
                class="header__auth"
              >
                <div>
                  <button
                    class="sc-bczRLJ avatar"
                  >
                    <img
                      class="avatar__image"
                      src="avatar.png"
                    />
                  </button>
                  <div
                    class="user-info "
                  >
                    <span
                      class="user-info__name"
                    >
                      TestName TestSurname
                    </span>
                    <ul
                      class="user-info__links"
                    >
                      <li
                        class="user-info__link-wrapper"
                      >
                        <a
                          class="user-info__link"
                          href="/favorites"
                          tabindex="-1"
                        >
                          Избранное
                        </a>
                      </li>
                      <li
                        class="user-info__link-wrapper"
                      >
                        <a
                          class="user-info__link"
                        >
                          Мои баллы
                        </a>
                      </li>
                      <li
                        class="user-info__link-wrapper"
                      >
                        <a
                          class="user-info__link"
                        >
                          Мои адреса доставки
                        </a>
                      </li>
                      <li
                        class="user-info__link-wrapper"
                      >
                        <a
                          class="user-info__link"
                        >
                          Пригласить друга
                        </a>
                      </li>
                    </ul>
                    <button
                      class="user-info__exit"
                      tabindex="-1"
                    >
                      Выход
                    </button>
                  </div>
                </div>
              </div>
              <a
                class="header__cart"
                href="/cart"
              >
                <svg
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Корзина
                  </title>
                  <path
                    d="M176 416C149.49 416 128 437.49 128 464S149.49 512 176 512S224 490.51 224 464S202.51 416 176 416ZM464 416C437.49 416 416 437.49 416 464S437.49 512 464 512S512 490.51 512 464S490.51 416 464 416ZM569.529 44.734C563.42 36.641 554.107 32 543.967 32H121.957L119.578 19.51C117.422 8.189 107.523 0 96 0H24C10.746 0 0 10.744 0 24C0 37.254 10.746 48 24 48H76.141L136.424 364.488C138.58 375.809 148.479 384 160.002 384H488C501.254 384 512 373.254 512 360C512 346.744 501.254 336 488 336H179.859L170.717 288H489.123C503.406 288 515.959 278.531 519.885 264.797L574.748 72.797C577.529 63.047 575.623 52.828 569.529 44.734ZM477.049 240H161.574L131.1 80H522.77L477.049 240Z"
                    fill="#FFF5E2"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <button
        aria-label="главное меню"
        class="header__burger-button "
      >
        <div
          class="header__burger"
        />
      </button>
      <div
        class="header__wrapper"
      >
        <div
          class="header__logo"
        >
          <a
            class="header__nav-link"
            href="/"
            title="главная страница"
          >
            <img
              alt="главная страница"
              class="header__nav-img"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="header__content "
        >
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item"
              >
                <a
                  class="header__nav-link"
                  href="/about"
                >
                  О нас
                </a>
              </li>
              <li
                class="header__nav-item"
              >
                <a
                  class="header__nav-link"
                  href="/deals"
                >
                  Акции
                </a>
              </li>
              <li
                class="header__nav-item"
              >
                <a
                  class="header__nav-link"
                  href="/blog"
                >
                  Блог
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="header__user"
          >
            <div
              class="header__auth"
            >
              <div>
                <button
                  class="sc-bczRLJ avatar"
                >
                  <img
                    class="avatar__image"
                    src="avatar.png"
                  />
                </button>
                <div
                  class="user-info "
                >
                  <span
                    class="user-info__name"
                  >
                    TestName TestSurname
                  </span>
                  <ul
                    class="user-info__links"
                  >
                    <li
                      class="user-info__link-wrapper"
                    >
                      <a
                        class="user-info__link"
                        href="/favorites"
                        tabindex="-1"
                      >
                        Избранное
                      </a>
                    </li>
                    <li
                      class="user-info__link-wrapper"
                    >
                      <a
                        class="user-info__link"
                      >
                        Мои баллы
                      </a>
                    </li>
                    <li
                      class="user-info__link-wrapper"
                    >
                      <a
                        class="user-info__link"
                      >
                        Мои адреса доставки
                      </a>
                    </li>
                    <li
                      class="user-info__link-wrapper"
                    >
                      <a
                        class="user-info__link"
                      >
                        Пригласить друга
                      </a>
                    </li>
                  </ul>
                  <button
                    class="user-info__exit"
                    tabindex="-1"
                  >
                    Выход
                  </button>
                </div>
              </div>
            </div>
            <a
              class="header__cart"
              href="/cart"
            >
              <svg
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Корзина
                </title>
                <path
                  d="M176 416C149.49 416 128 437.49 128 464S149.49 512 176 512S224 490.51 224 464S202.51 416 176 416ZM464 416C437.49 416 416 437.49 416 464S437.49 512 464 512S512 490.51 512 464S490.51 416 464 416ZM569.529 44.734C563.42 36.641 554.107 32 543.967 32H121.957L119.578 19.51C117.422 8.189 107.523 0 96 0H24C10.746 0 0 10.744 0 24C0 37.254 10.746 48 24 48H76.141L136.424 364.488C138.58 375.809 148.479 384 160.002 384H488C501.254 384 512 373.254 512 360C512 346.744 501.254 336 488 336H179.859L170.717 288H489.123C503.406 288 515.959 278.531 519.885 264.797L574.748 72.797C577.529 63.047 575.623 52.828 569.529 44.734ZM477.049 240H161.574L131.1 80H522.77L477.049 240Z"
                  fill="#FFF5E2"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
